reTurn/client/TurnAsyncSocket.cxx

```diff
@@ -1114,6 +1115,33 @@ TurnAsyncSocket::sendToRemotePeer(RemotePeer& remotePeer, const std::shared_ptr<
    }
 }
 
+void
+TurnAsyncSocket::setSoftware(const std::string &software)
```

Comment:
This API is not thread safe.  I would recommend that you follow the pattern established by the methods setLocalPassword and doSetLocalPassword.  This shows how to dispatch the setting of mSoftware to the thread where the string is used, and ensures you can change it at anytime in a thread safe manner.